zrml/prediction-markets/src/lib.rs

```diff
@@ -1400,14 +1413,36 @@ mod pallet {
                     },
                 );
 
+                if let Ok(weight) = close {
+                    total_weight = total_weight.saturating_add(weight);
+                } else {
+                    // charge weight for the worst case
+                    total_weight = total_weight.saturating_add(
+                        T::WeightInfo::market_status_manager(CacheSize::get(), CacheSize::get()),
```

Comment:
Of course, the worst case for `market_status_manager` is not `CacheSize::get()`; it's a very large difference between `current_time_frame` and `last_time_frame`, and each time frame holds `CacheSize::get()`. The larger the gap between the time frames, the larger the weight.

I don't think we should do this right now, but can you open an issue that we need to basically delay status changes until the next block if there are too many markets to handle? I would do it like this:

- Set a maximum weight for status changes `m`
- Benchmark the weight of a single status change `w`; if we need to handle more markets than `m / w` markets, we need to delay the status change until the next block

This basically boils down to having a priority system, where the weight `m` needs to be distributed between a couple of tasks (status changes, garbage collecting) and opening/closing is more important than resolving, which in turn is more important than garbage collection.

This shouldn't be difficult to implement, we just need to find the time to do it.

Reply:
I see your point. I added your comment in this issue https://github.com/zeitgeistpm/zeitgeist/issues/817